### PR TITLE
Also name non-chunk files with "_temp.avro" suffix during the writing…

### PR DESCRIPTION
**Pull request by cyulysses-corp**
… process, then remove "_temp" after the process completes (chunk files already behave this way). This allows the program to distinguish completed record files versus incomplete ones (usually due to interruption). It is a prerequisite for "recovery run" feature.

Conducted manual tests already, during which "_temp.avro" files would appear and then got renamed without _temp after the writing finished.

### CODE REVIEW

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java [hunk 1]
```diff
@@ -151,16 +151,23 @@ private void executeScriptOneSwoop(
       Schema schema,
       DataEntityManager dataEntityManager)
       throws SQLException, IOException {
+    String fileSuffix = dataEntityManager.isResumable() ? "_temp.avro" : ".avro";
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:154]: ".avro" as a string literal is used 4 times in the file. Consider making it `static final`
The same suggestion, but with the different justification, applies to "_temp.avro" – it's not a method specific and should be moved as well. Also, I expect it anyway will be needed for the recovery functionality

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java [hunk 2]
```diff
@@ -151,16 +151,23 @@ private void executeScriptOneSwoop(
...
-            schema, dataEntityManager.getEntityOutputStream(scriptName + ".avro"))) {
+            schema, dataEntityManager.getEntityOutputStream(scriptName + fileSuffix))) {
       scriptRunner.executeScriptToAvro(connection, script, schema, dumper::add);
     } catch (IOException | SQLException e) {
       throw e;
     } catch (Exception e) {
       // Cannot happen.
       throw new IllegalStateException("Got unexpected exception.", e);
     }
+    if (dataEntityManager.isResumable()) {
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:165]: Please, add tests for this part

**Review by frsv — COMMENTED** [review]
I feel the code becomes more complex

UPD: I forgot to remove the line above, it doesn't make any sense alone. For the sake of history, will leave it as it is

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java (hunk 1 above)
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java]: Actually "_temp" or ".avro" will not be used individually for the recovery functionality (rather as part of a larger regex pattern) but I don't think it hurts to give them a special place.

**Comment by cyulysses-corp** [comment]
> I feel the code becomes more complex

Unfortunately it is, a little bit. There was a plan to refactor the structure to make everything more readable but was shelved in favour of another tool.
The good news is that there won't be many more added features in the future for this tool, so this is likely as complex as it gets.

**Review by frsv — APPROVED** [review]

**Pull request merged**